internal/controller/ipaddress_controller.go

```diff
@@ -101,6 +101,15 @@ func (r *IpAddressReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{}, nil
 	}
 
+	// if PreserveIpInNetbox flag is false then register finalizer if not yet registered
```

Comment:
Currently the finalizer won't be removed if the deletion of the ipaddress in NetBox fails. This also includes the case where the resource doesn't exist.

Adding the finalizer before the resource was created in NetBox could lead to resources that cannot be deleted by the operator. Should we edit the reconcile loop to only try the deletion if Status.IpAddressId is not `0` and to not found errors from `NetboxClient.DeleteIpAddress`.